include/tvm/requirements/abstract/SingleSolvingRequirement.h

```diff
@@ -44,19 +48,57 @@ namespace abstract
     *
     * This is a base class for the sole purpose of conveniency.
     */
-  template<typename T>
-  class SingleSolvingRequirement
+  template<typename T, bool Lightweight = true>
+  class SingleSolvingRequirement : public std::conditional_t<Lightweight, std::monostate, internal::CallbackManager>
   {
+    using Base = std::conditional_t<Lightweight, std::monostate, internal::CallbackManager>;
+
   public:
+    //template<bool b = Lightweight, typename = std::enable_if_t<b, int> >
+    //SingleSolvingRequirement(const SingleSolvingRequirement&) = default;
+    //template<bool b = Lightweight, typename = std::enable_if_t<b, int> >
+    //SingleSolvingRequirement& operator=(const SingleSolvingRequirement&) = default;
+    //template<bool b = Lightweight, typename = std::enable_if_t<!b, int> >
+    //SingleSolvingRequirement(const SingleSolvingRequirement&) = delete;
+    //template<bool b = Lightweight, typename = std::enable_if_t<!b, int> >
+    //SingleSolvingRequirement& operator=(const SingleSolvingRequirement&) = delete;
+
+    /** Get the current value. */
     const T& value() const { return value_; }
+
+    /** Change the current value */
+    void value(const T& val)
+    {
+      value_ = val;
+      default_ = false;
+      if constexpr (!Lightweight)
+      {
+        Base::run();
```

Comment:
Here, this about changing the value of the requirement. So for an `AnisotropicWeight`, this would mean changing the value of the vector.

Resizing a function/task that is linked to an `AnisotropicWeight` is another topic and it will be tricky to resize the `AnisotropicWeight` automatically: how do you decide the values of he resized vector? The saving grace could be that we are going for delayed reactions to events, so that the "user" would have time to resize both the function and the `AnisotropicWeight` before we process the changes. Otherwise, I think we should throw when resizing in such case.

Reply:
So we will prevent all function resizing capabilities? (I think it's ok) Otherwise I think it will be confusing if you can resize the function until you associate an anisotropic weight to it and suddenly it becomes impossible (and the same concern applies when task dynamics with dimensional values enter in play as well)

Another possibility is that the resize event will have to carry new dimensional quantities?

Reply:
It would be nice to have resizing capabilities for function like the collision, but it will definitely be one of the most intricate parts of the even handling. I guess we can avoid confusion with a clear exception system.

You're right that resizing can also be a problem with some tasks dynamics, and more generally it will be a problem with piping computation nodes when there is not an obvious way to resize parameters and data from a node.

Let's start without function resizing and see later what we can do about it.

Reply:
> It would be nice to have resizing capabilities for function like the collision, but it will definitely be one of the most intricate parts of the even handling. I guess we can avoid confusion with a clear exception system.

This was indeed one of the scenario we discussed but in the mc_rtc/TVM branch I've actually implemented collision as one dimensional functions because there's actually zero computations that are shared between the dimensions. The only thing I am missing now to reduce the collision constraint burden on the solver is to remove the constraint  while the graph is being updated (maybe extra hard?) but I can also work-around the limitation and do the removal decision before the graph update which will work soon enough :)

> Let's start without function resizing and see later what we can do about it.

Yes. We'll see if the need really arises or not